packages/mdc-dialog/foundation.js

```diff
@@ -196,11 +196,14 @@ class MDCDialogFoundation extends MDCFoundation {
    * @private
    */
   handleClick(evt) {
-    const action = this.adapter_.getActionFromEvent(evt);
-    if (action) {
-      this.close(action);
-    } else if (this.adapter_.eventTargetHasClass(evt.target, cssClasses.SCRIM) && this.scrimClickAction_) {
+    // Check for scrim click first since it doesn't require querying ancestors
+    if (this.adapter_.eventTargetHasClass(evt.target, cssClasses.SCRIM) && this.scrimClickAction_) {
```

Comment:
This is bit confusing. Instead of setting scrim action with `setScrimClickAction` `closeOnBackgroundClick(boolean)` would have more intuitive. WDYT?

Reply:
1. Scrim is the word that the [spec](https://material.io/design/components/dialogs.html#anatomy) uses to refer to this element.
2. This API doesn't control a boolean, it controls a string. You can assign any string to the action in case you want to distinguish it for some reason, but both the escape key and scrim click default to reporting `close` action.

This is implemented for flexibility with widely varying use cases of dialogs.

Reply:
Inspired by @patrickrodee in #3576, I added explicit comparisons here to clarify how these variables work.